internal/render/pod_test.go

```diff
@@ -543,6 +544,82 @@ func TestCheckPodStatus(t *testing.T) {
 	}
 }
 
+func TestPodLastRestart(t *testing.T) {
+	uu := map[string]struct {
+		containerStatuses []v1.ContainerStatus
+		expected          metav1.Time
+	}{
+		"no-restarts": {
+			containerStatuses: []v1.ContainerStatus{
+				{
+					Name:                 "c1",
+					LastTerminationState: v1.ContainerState{},
+				},
+			},
+			expected: metav1.Time{},
+		},
+		"single-container-restart": {
+			containerStatuses: []v1.ContainerStatus{
+				{
+					Name: "c1",
+					LastTerminationState: v1.ContainerState{
+						Terminated: &v1.ContainerStateTerminated{
+							FinishedAt: metav1.Time{Time: testTime()},
+						},
+					},
+				},
+			},
+			expected: metav1.Time{Time: testTime()},
+		},
+		"multiple-container-restarts": {
+			containerStatuses: []v1.ContainerStatus{
+				{
+					Name: "c1",
+					LastTerminationState: v1.ContainerState{
+						Terminated: &v1.ContainerStateTerminated{
+							FinishedAt: metav1.Time{Time: testTime().Add(-1 * time.Hour)},
+						},
+					},
+				},
+				{
+					Name: "c2",
+					LastTerminationState: v1.ContainerState{
+						Terminated: &v1.ContainerStateTerminated{
+							FinishedAt: metav1.Time{Time: testTime()},
+						},
+					},
+				},
+			},
+			expected: metav1.Time{Time: testTime()},
+		},
+		"mixed-termination-states": {
+			containerStatuses: []v1.ContainerStatus{
+				{
+					Name:                 "c1",
+					LastTerminationState: v1.ContainerState{},
+				},
+				{
+					Name: "c2",
+					LastTerminationState: v1.ContainerState{
+						Terminated: &v1.ContainerStateTerminated{
+							FinishedAt: metav1.Time{Time: testTime()},
+						},
+					},
+				},
+			},
+			expected: metav1.Time{Time: testTime()},
+		},
+	}
+
+	var p render.Pod
+	for k := range uu {
+		u := uu[k]
```

Comment:
No longer necessary you can now just range over the map.

Reply:
Nice, will update it also